06 - Type Ahead/index-START.html

```diff
@@ -15,8 +15,68 @@
     </ul>
   </form>
 <script>
-const endpoint = 'https://gist.githubusercontent.com/Miserlou/c5cd8364bf9b2420bb29/raw/2bf258763cdddd704f8ffd3ea9a3e81d25e2c6f6/cities.json';
+  //// global variables
+  const endpoint = 'https://gist.githubusercontent.com/Miserlou/c5cd8364bf9b2420bb29/raw/2bf258763cdddd704f8ffd3ea9a3e81d25e2c6f6/cities.json';
+  let cities = [];
+  let searchTerm = '';
 
+  //// selectors
+  const form = document.querySelector('.search-form');
+  const input = document.querySelector('.search');
+  const suggestions = document.querySelector('.suggestions');
+
+  //// callbacks
+  // fetchData() is async if we decide to also fetch on change or on submit for example
+  const fetchData = async (URL) => {
+    try {
+      const res = await fetch(URL);
+      const data = await res.json();
+      cities = [...data];
+    } catch (error) {
+      const isError = () => {
```

Comment:
why do you need to define a function for this?

Reply:
In this case I actually do not need to use a function. Nonetheless, I like to use functions to keep my code clean. Also, fetching is actually one of the best cases to use a reusable component/function. It does not cost too many extra lines of code and the advantage is that you can always re-use it anywhere in your app. Most of the time you just need to re-call the function but with a different endpoint/URL and options.

Reply:
mmm, interesting. In general I agree. Reusing is a good goal. Also sometimes it is nice to add a function just for structural purposes. In this case however I am not sure if I agree by adding a function. here the variable `isError` is defined in a deep scope, so we can not reuse it. I think the extra code makes it harder to read. maybe try to remove it and compare both versions. then pick which version you think is better.

Reply:
I tried and I opted for the promise chaining ... async/await without a function looks naked :)

Reply:
Ops! yes I misunderstood! I left the promise chaining but removed the `isError` function now! look better and makes totally sense.

Reply:
super cool.